### PR TITLE
fix the manifestWork in hosted cluster ns is not deleted in hosted mode

### PR DESCRIPTION
**Pull request by zhiweiyin318**
Signed-off-by: Zhiwei Yin <zyin@redhat.com>

### CODE REVIEW

File: pkg/addonmanager/controllers/agentdeploy/utils.go [hunk 1]
```diff
@@ -242,10 +242,8 @@ func (b *addonWorksBuilder) BuildDeployWorks(addonWorkNamespace string,
 	var owner *metav1.OwnerReference
 	installMode, _ := constants.GetHostedModeInfo(addon.GetAnnotations())
 
-	// only set addon as the owner of works in default mode. should not set owner in hosted mode.
-	if installMode == constants.InstallModeDefault {
-		owner = metav1.NewControllerRef(addon, addonapiv1alpha1.GroupVersion.WithKind("ManagedClusterAddOn"))
-	}
+	// The manifestWork deployed in cluster ns should have the owner in hosted mode.
+	owner = metav1.NewControllerRef(addon, addonapiv1alpha1.GroupVersion.WithKind("ManagedClusterAddOn"))
```
Comment (qiujian16) [inline pkg/addonmanager/controllers/agentdeploy/utils.go:248]: will it owner cross namespace?

File: pkg/addonmanager/controllers/agentdeploy/utils.go [hunk 2]
```diff
@@ -427,7 +425,7 @@ func newAddonWorkObjectMeta(namePrefix, addonName, addonNamespace, workNamespace
 				constants.AddonLabel: addonName,
 			},
 		}
-		if owner != nil {
+		if addonNamespace == workNamespace && owner != nil {
```
Comment (qiujian16) [inline pkg/addonmanager/controllers/agentdeploy/utils.go:433]: I see what is going on here, but we need to doc this.

Comment (zhiweiyin318) [reply pkg/addonmanager/controllers/agentdeploy/utils.go:433]: added doc here . the owner is only added to manifestWork deployed in the managed cluster ns.

File: pkg/addonmanager/controllers/agentdeploy/utils.go (hunk 1 above)
Comment (zhiweiyin318) [reply pkg/addonmanager/controllers/agentdeploy/utils.go:248]: added doc here . the owner is only added to manifestWork deployed in the managed cluster ns.

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**